lib/matplotlib/backends/backend_pdf.py

```diff
@@ -1700,8 +1700,19 @@ def _writeImg(self, data, id, smask=None):
                 # Convert to indexed color if there are 256 colors or fewer
                 # This can significantly reduce the file size
                 num_colors = len(img_colors)
-                img = img.convert(mode='P', dither=Image.NONE,
-                                  palette=Image.ADAPTIVE, colors=num_colors)
+                # These constants were converted to IntEnums and deprecated in
+                # Pillow 9.2
+                dither = (
```

Comment:
I know this style is odd, but these names a very long and this code is very indented.

I also tried

```python
                dither = (
                    Image.Dither.NONE
                    if hasattr(Image, 'Dither')
                    else Image.NONE
                )
```
and liked that worse.

also tried doing this in-line in the function call, but that was also (subjectively) worse to my eye.